tests/virtualization/universal/prepare_guests.pm

```diff
@@ -60,6 +59,23 @@ sub create_profile {
     return autoinst_url . "/files/$vm_name.xml";
 }
 
+sub gen_osinfo {
+    my ($vm_name) = @_;
+    my $h_version = get_var("VERSION") =~ s/-SP/./r;
+    my $g_version = $vm_name =~ /sp/ ? $vm_name =~ s/\D*(\d+)sp(\d)\D*/$1.$2/r : $vm_name =~ s/\D*(\d+)\D*/$1/r;
+    my $info_op = $h_version > 15.2 ? "--osinfo" : "--os-variant";
+    my $info_val = $g_version > 12.5 ? $vm_name =~ s/HVM|PV//r =~ s/sles/sle/r : $vm_name =~ s/PV|HVM//r;
+    if ($h_version == 12.3) {
+        $info_val = "sle15-unknown" if ($g_version > 15.1);
+        $info_val = "sles12-unknown" if ($g_version == 12.5);
+    } elsif ($h_version == 12.4) {
+        $info_val = "sle15-unknown" if ($g_version > 15.2);
+    } elsif ($h_version == 15) {
+        $info_val = "sle15-unknown" if ($g_version > 15.1);
+    }
+    return "$info_op $info_val";
```

Comment:
Since the logic of OS version judgment is a little complicated, suggest to add description here to make it more legible.

Reply:
Thanks for the comments, updated code according it.